python/ray/data/_internal/planner/exchange/pull_based_shuffle_task_scheduler.py

```diff
@@ -47,7 +48,15 @@ def execute(
         shuffle_map = cached_remote_fn(self._exchange_spec.map)
         shuffle_reduce = cached_remote_fn(self._exchange_spec.reduce)
 
-        map_bar = ProgressBar("Shuffle Map", total=input_num_blocks)
+        sub_progress_bar_dict = ctx.sub_progress_bar_dict
+        should_close_bar = True
+        bar_name = "Shuffle Map"
+        if sub_progress_bar_dict is not None:
+            assert bar_name in sub_progress_bar_dict, sub_progress_bar_dict
+            map_bar = sub_progress_bar_dict[bar_name]
+            should_close_bar = False
+        else:
+            map_bar = ProgressBar(bar_name, position=0, total=input_num_blocks)
```

Comment:
Is it possible to get rid of these if-else branches? they are a little bit ugly. In which cases, the `sub_progress_bar_dict` is None? Would be great to unify the code logic.